sirplatform.h

```diff
@@ -164,6 +164,7 @@ int pthread_getname_np(pthread_t thread, char* buffer, size_t length);
 # include <errno.h>
 # include <stdarg.h>
 # include <stdbool.h>
+# include <stddef.h>
```

Comment:
I don't understand the purpose of this include at this time. Can you explain?

Reply:
The reference code wants it, (apparently) because that's where the [C11 `size_t` definition lives](https://en.cppreference.com/w/c/types).  I guess it's getting pulled in for us via string.h and/or uchar.h and/or wchar.h implicitly, so it might not be strictly required, but since it's C11 I didn't think it would hurt, and I didn't want to have the strl* files include anything themselves.

Reply:
Check out what cppreference.com says about it: https://en.cppreference.com/w/c/types/size_t

Looks like there's a billion ways to get it, and it claims for C11, it's in `uchar.h`...

Reply:
Since we aren't using Unicode strings, seems overkill to pull 100+ defintions in ... the entire stddef everywhere is something tiny like:

```c
#ifndef _STDDEF_H
#define _STDDEF_H
#if __cplusplus >= 201103L
#define NULL nullptr
#elif defined(__cplusplus)
#define NULL 0L
#else
#define NULL ((void*)0)
#endif
#define __NEED_ptrdiff_t
#define __NEED_size_t
#define __NEED_wchar_t
#if __STDC_VERSION__ >= 201112L || __cplusplus >= 201103L
#define __NEED_max_align_t
#endif
#include <bits/alltypes.h>
#if __GNUC__ > 3
#define offsetof(type, member) __builtin_offsetof(type, member)
#else
#define offsetof(type, member) ((size_t)( (char *)&(((type *)0)->member) - (char *)0 ))
#endif
#endif
```

I don't mind to try leave it out though, and add it only if any platform gives us a problem.